src/config/Config.cpp

```diff
@@ -89,101 +90,149 @@ void Config::reset(bool lock) {
   if (lock) Config::unlock();
 }
 
-Config::InternetConnectionType Config::getInternetConnectionType(bool lock) {
-  return Config::internetConnectionType;
+Config::InternetConnectionType Config::getInternetConnectionType(bool lock /* = true */) {
+  if (lock) Config::lock();
+  InternetConnectionType value = Config::internetConnectionType;
+  if (lock) Config::unlock();
+  return value;
```

Comment:
I like that someone noticed this and changed to proper way of handling locking :)